packages/exporter-trace-otlp-http/src/platform/browser/util.ts

```diff
@@ -51,9 +51,17 @@ export function sendWithXhr(
   body: string,
   url: string,
   headers: Record<string, string>,
+  exporterTimeout: number,
   onSuccess: () => void,
   onError: (error: otlpTypes.OTLPExporterError) => void
 ): void {
+  let reqIsDestroyed: boolean;
+
+  const exporterTimer = setTimeout(() => {
+    reqIsDestroyed = true;
+    xhr.abort();
+  }, exporterTimeout);
+
   const xhr = new XMLHttpRequest();
   xhr.open('POST', url);
```

Comment:
As long as you are not performing a synchronous request, XHR has a `timeout` field. Why not just ust that and then listen to the onTimeout?